src/mlpack/core/kernels/hyperbolic_tangent_kernel.hpp

```diff
@@ -64,12 +64,12 @@ class HyperbolicTangentKernel
   //! Get scale factor.
   double Scale() const { return scale; }
   //! Modify scale factor.
-  double& Scale() { return scale; }
+  void Scale(const double scale) { this->scale = scale; }
```

Comment:
Any advantage to moving to this method of setting the private members?

Reply:
In this case, no---but in general this style could be used to validate parameters when they are set.  In some cases this could also allow you to set some other internal members that depend on the parameter being modified.

Throughout the codebase there is some inconsistency here; actually the form this was before doesn't make any sense... it doesn't provide any protection, doesn't allow validation of parameters, and is functionally not different at all from just making the private member public.  You can blame 2010 Ryan for not thinking this through fully, this was my error for picking this style a very long time ago.  The `void Param(param)` style is more sensible.